### PR TITLE
[#1046] Refactor duplicate code

### LINKED ISSUE #1046: Duplicate code in v_summary.js
There is duplicate code in v_summary.js. 

Screen Shot 2020-02-03 at 4 42 08 AM

Screen Shot 2020-02-03 at 4 41 53 AM

### PR DESCRIPTION
**Pull request by yash-chowdhary**
Fixes #1046 

```
There is an instance of code duplication in 'v_summary.js'. 
Specifically, the definition of a function that is passed to the comparator 
has been duplicated in the case of grouping by repos and by authors.

Let's extract this duplicated portion of code into a helper function so that
it may be referenced by the comparator.
```

### CODE REVIEW

**Comment by yash-chowdhary** [comment]
Ready for review.

File: frontend/src/static/js/v_summary.js [hunk 1]
```diff
@@ -3,9 +3,20 @@ window.dismissTab = function dismissTab(node) {
...
+window.comparator = (fn, sortingOption = '') => function compare(a, b) {
+  let a1;
+  let b1;
+  if (sortingOption) {
+    a1 = fn(a, sortingOption).toLowerCase
+    ? fn(a, sortingOption).toLowerCase()
+    : fn(a, sortingOption);
+    b1 = fn(b, sortingOption).toLowerCase
+    ? fn(b, sortingOption).toLowerCase()
+    : fn(b, sortingOption);
```
Comment (jylee-git) [inline frontend/src/static/js/v_summary.js]: four leading spaces for multiline ternary operators

File: frontend/src/static/js/v_summary.js [hunk 2]
```diff
@@ -792,6 +793,13 @@ window.vSummary = {
     getGroupTotalContribution(group) {
       return group.reduce((accContribution, user) => accContribution + user.totalCommits, 0);
     },
+
+    sortingHelper(element, sortingOption) {
+      if (sortingOption === 'totalCommits' || sortingOption === 'variance') {
+        return element.reduce(this.getGroupCommitsVariance, 0);
+      }
+      return element[0][sortingOption];
+    },
```
Comment (jylee-git) [inline frontend/src/static/js/v_summary.js]: Will a ternary operator be better here?

Comment (yash-chowdhary) [reply frontend/src/static/js/v_summary.js]: Made the change 👍

**Comment by jamessspanggg** [comment]
@fzdy1914 need your review here.

**Pull request merged**